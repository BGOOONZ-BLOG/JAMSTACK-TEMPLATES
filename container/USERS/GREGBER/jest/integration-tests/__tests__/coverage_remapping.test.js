/**
 * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow
 */

'use strict';

const {readFileSync} = require('fs');
const path = require('path');
const {cleanup, run} = require('../Utils');
const runJest = require('../runJest');

const dir = path.resolve(__dirname, '../coverage-remapping');
const coverageDir = path.join(dir, 'coverage');

beforeAll(() => {
  cleanup(coverageDir);
});

it('maps code coverage against original source', () => {
  run('yarn', dir);
  const result = runJest(dir, ['--coverage', '--no-cache']);

  expect(result.status).toBe(0);

  const coverageMapFile = path.join(coverageDir, 'coverage-final.json');
  const coverageMap = JSON.parse(readFileSync(coverageMapFile, 'utf-8'));

  // reduce absolute paths embedded in the coverage map to just filenames
  Object.keys(coverageMap).forEach(filename => {
    coverageMap[filename].path = path.basename(coverageMap[filename].path);
    delete coverageMap[filename].hash;
    coverageMap[path.basename(filename)] = coverageMap[filename];
    delete coverageMap[filename];
  });
  expect(coverageMap).toMatchSnapshot();
});
