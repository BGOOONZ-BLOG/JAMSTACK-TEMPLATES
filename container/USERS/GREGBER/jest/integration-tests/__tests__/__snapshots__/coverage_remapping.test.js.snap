// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maps code coverage against original source 1`] = `
Object {
  "covered.ts": Object {
    "b": Object {
      "0": Array [
        1,
        0,
      ],
      "1": Array [
        1,
        0,
      ],
      "2": Array [
        1,
        0,
        0,
      ],
      "3": Array [
        1,
        0,
      ],
    },
    "branchMap": Object {
      "0": Object {
        "loc": Object {
          "end": Object {
            "column": 9,
            "line": 5,
          },
          "start": Object {
            "column": 8,
            "line": 5,
          },
        },
        "locations": Array [
          Object {
            "end": Object {
              "column": 9,
              "line": 5,
            },
            "start": Object {
              "column": 8,
              "line": 5,
            },
          },
          Object {
            "end": Object {
              "column": 9,
              "line": 6,
            },
            "start": Object {
              "column": 8,
              "line": 6,
            },
          },
        ],
        "type": "cond-expr",
      },
      "1": Object {
        "loc": Object {
          "end": Object {
            "column": 37,
            "line": 7,
          },
          "start": Object {
            "column": 36,
            "line": 7,
          },
        },
        "locations": Array [
          Object {
            "end": Object {
              "column": 37,
              "line": 7,
            },
            "start": Object {
              "column": 36,
              "line": 7,
            },
          },
          Object {
            "end": Object {
              "column": 41,
              "line": 7,
            },
            "start": Object {
              "column": 40,
              "line": 7,
            },
          },
        ],
        "type": "cond-expr",
      },
      "2": Object {
        "loc": Object {
          "end": Object {
            "column": 33,
            "line": 8,
          },
          "start": Object {
            "column": 29,
            "line": 8,
          },
        },
        "locations": Array [
          Object {
            "end": Object {
              "column": 33,
              "line": 8,
            },
            "start": Object {
              "column": 29,
              "line": 8,
            },
          },
          Object {
            "end": Object {
              "column": 41,
              "line": 8,
            },
            "start": Object {
              "column": 37,
              "line": 8,
            },
          },
          Object {
            "end": Object {
              "column": 50,
              "line": 8,
            },
            "start": Object {
              "column": 45,
              "line": 8,
            },
          },
        ],
        "type": "binary-expr",
      },
      "3": Object {
        "loc": Object {
          "end": Object {
            "column": 42,
            "line": 9,
          },
          "start": Object {
            "column": 32,
            "line": 9,
          },
        },
        "locations": Array [
          Object {
            "end": Object {
              "column": 42,
              "line": 9,
            },
            "start": Object {
              "column": 32,
              "line": 9,
            },
          },
          Object {
            "end": Object {
              "column": 55,
              "line": 9,
            },
            "start": Object {
              "column": 45,
              "line": 9,
            },
          },
        ],
        "type": "cond-expr",
      },
    },
    "f": Object {
      "0": 1,
      "1": 0,
      "2": 0,
    },
    "fnMap": Object {
      "0": Object {
        "decl": Object {
          "end": Object {
            "column": 28,
            "line": 3,
          },
          "start": Object {
            "column": 9,
            "line": 3,
          },
        },
        "loc": Object {
          "end": Object {
            "column": 1,
            "line": 12,
          },
          "start": Object {
            "column": 49,
            "line": 3,
          },
        },
        "name": "difference",
      },
      "1": Object {
        "decl": Object {
          "end": Object {
            "column": 37,
            "line": 9,
          },
          "start": Object {
            "column": 32,
            "line": 9,
          },
        },
        "loc": Object {
          "end": Object {
            "column": 42,
            "line": 9,
          },
          "start": Object {
            "column": 32,
            "line": 9,
          },
        },
        "name": "(anonymous_1)",
      },
      "2": Object {
        "decl": Object {
          "end": Object {
            "column": 50,
            "line": 9,
          },
          "start": Object {
            "column": 45,
            "line": 9,
          },
        },
        "loc": Object {
          "end": Object {
            "column": 55,
            "line": 9,
          },
          "start": Object {
            "column": 45,
            "line": 9,
          },
        },
        "name": "(anonymous_2)",
      },
    },
    "path": "covered.ts",
    "s": Object {
      "0": 1,
      "1": 1,
      "2": 1,
      "3": 1,
      "4": 1,
      "5": 0,
      "6": 0,
      "7": 1,
    },
    "statementMap": Object {
      "0": Object {
        "end": Object {
          "column": 1,
          "line": 12,
        },
        "start": Object {
          "column": 0,
          "line": 3,
        },
      },
      "1": Object {
        "end": Object {
          "column": 9,
          "line": 6,
        },
        "start": Object {
          "column": 29,
          "line": 4,
        },
      },
      "2": Object {
        "end": Object {
          "column": 41,
          "line": 7,
        },
        "start": Object {
          "column": 29,
          "line": 7,
        },
      },
      "3": Object {
        "end": Object {
          "column": 50,
          "line": 8,
        },
        "start": Object {
          "column": 29,
          "line": 8,
        },
      },
      "4": Object {
        "end": Object {
          "column": 55,
          "line": 9,
        },
        "start": Object {
          "column": 25,
          "line": 9,
        },
      },
      "5": Object {
        "end": Object {
          "column": 42,
          "line": 9,
        },
        "start": Object {
          "column": 38,
          "line": 9,
        },
      },
      "6": Object {
        "end": Object {
          "column": 55,
          "line": 9,
        },
        "start": Object {
          "column": 51,
          "line": 9,
        },
      },
      "7": Object {
        "end": Object {
          "column": 17,
          "line": 11,
        },
        "start": Object {
          "column": 4,
          "line": 11,
        },
      },
    },
  },
}
`;
